bundler/lib/bundler/rubygems_ext.rb

```diff
@@ -277,6 +282,10 @@ def normalized_linux_version_ext
         without_gnu_nor_abi_modifiers
       end
     end
+
+    if RUBY_ENGINE == "truffleruby" && !defined?(REUSE_AS_BINARY_ON_TRUFFLERUBY)
+      REUSE_AS_BINARY_ON_TRUFFLERUBY = %w[libv8 libv8-node sorbet-static].freeze
+    end
```

Comment:
Later I will propose that truffleruby moves this constant to the `defaults/truffleruby` file, so that eventually we no longer need to keep it ourselves. We currently need to keep it ourselves for our own tests, and to work properly with upgraded copies of RubyGems that don't include it like the "blessed copy" of RubyGems included with truffleruby does.

Reply:
Should I do a PR to truffleruby to have
```ruby
class Gem::Platform
  remove_const :REUSE_AS_BINARY_ON_TRUFFLERUBY if const_defined?(:REUSE_AS_BINARY_ON_TRUFFLERUBY)
  REUSE_AS_BINARY_ON_TRUFFLERUBY = %w[libv8 libv8-node sorbet-static]
end
```
in `defaults/truffleruby` then?
I can do that soon.

Or where you thinking of something else?

Reply:
Yup, that's what I was thinking.

Reply:
Depending on https://github.com/rubygems/rubygems/pull/6238#issuecomment-1372313641 we might need to remove the constant

Reply:
Yep, you shouldn't need `remove_const` because of that.